File: src/server/src/main/java/com/project/domain/valueobjects/FullName.java
package com.project.domain.valueobjects;

import jakarta.persistence.Embeddable;

@Embeddable
public class FullName {
    private String firstName;
    private String lastName;

    private FullName(String firstName, String lastName) {
        this.firstName = firstName;
        this.lastName = lastName;
    }

    protected FullName() {}

    public String getFullName() {
        return String.format("%s %s", this.firstName, this.lastName);
    }

    public static FullName from(String firstName, String lastName) {
        return new FullName(firstName,lastName);
    }

    public String getFirstName() {
        return firstName;
    }

    public String getLastName() {
        return lastName;
    }
}

File: src/server/src/main/java/com/project/domain/valueobjects/FullName.kt
package com.project.domain.valueobjects

import jakarta.persistence.Embeddable

@Embeddable
class FullName {
    var firstName: String? = null
        private set
    var lastName: String? = null
        private set

    private constructor(firstName: String, lastName: String) {
        this.firstName = firstName
        this.lastName = lastName
    }

    protected constructor()

    val fullName: String
        get() = String.format("%s %s", this.firstName, this.lastName)

    companion object {
        fun from(firstName: String, lastName: String): FullName {
            return FullName(firstName, lastName)
        }
    }
}
